src/components/TraceDetail.tsx

```diff
@@ -256,20 +263,17 @@ function TraceDetail({
   }, [result.isSuccess, result.data]);
 
   const traceDurationInMiliseconds = React.useMemo(() => {
-    if (!result.isSuccess || result.data.length === 0) {
+    if (!result.isSuccess || result.data.spans.length === 0) {
       return 0;
     }
-    const rootSpan = result.data[0];
-    return (
-      mkMilisecondsFromNanoSeconds(rootSpan.endTimeUnixNano) - mkMilisecondsFromNanoSeconds(rootSpan.startTimeUnixNano)
-    );
+    return result.data.durationMs;
   }, [result.isSuccess, result.data]);
 
   const traceStartTimeInMiliseconds = React.useMemo(() => {
```

Comment:
We have this value from the props. I think this is a remnant of earlier prototypes.